net/src/headers/mod.rs

```diff
@@ -166,7 +166,12 @@ impl Net {
 }
 
 impl Transport {
-    pub(crate) fn update_checksum(&mut self, net: &Net, payload: impl AsRef<[u8]>) {
+    pub(crate) fn update_checksum(
+        &mut self,
+        net: &Net,
+        embedded_headers: Option<&EmbeddedHeaders>,
+        payload: impl AsRef<[u8]>,
+    ) {
         match (net, self) {
```

Comment:
I'm not really a fan of this approach but it gets the job done.  We probably need a better way to just handle structured payloads and serialize them at the end.  Not worth doing now though.

Reply:
Not a fan, either. I don't see a cleaner, non-trivial solution. I can't find a good model that allows to efficiently consider inner headers both as some headers layer, and as payload for the outer headers :disappointed:.